open_academy/models/session.py

```diff
@@ -1,13 +1,48 @@
-from odoo import models, fields
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
 
 
 class Session(models.Model):
     _name = 'session'
     _description = 'Session'
     course_id = fields.Many2one('course', required=True)
-    instructor_id = fields.Many2one('res.partner')
+    instructor_id = fields.Many2one('res.partner', domain=(
+        "['|',"
+        "('is_instructor', '=', 'True'),"
+        "('category_id.name', 'like', 'Teacher'),]"),
+    )
```

Comment:
Again, looks bad, Check pythons PEP to best code practices, https://www.python.org/dev/peps/

```suggestion
    instructor_id = fields.Many2one(
        "res.partner",
        domain=("['|', ('is_instructor', '=', 'True'), ('category_id.name', 'like', 'Teacher')]"),
    )
```